### PR TITLE
Calculate normals for facets more efficiently

### LINKED ISSUE #5379: ~70% Performance speedup for polygon_mesh_to_nef_3
## Issue Details

I have been investigating the performance of nef polyhedron construction, and have noticed the calculation of normals using normal_newell_3, while this approach is suitable for convex/concave facets with an arbitrary number of vertices, I think it is suboptimal for triangular facets. The function CGAL::orthogonal_vector is much better for this task, and we can easily switch between these two methods based on the number of edges/vertices.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux 64bit
* Compiler: clang++ 
* Release or debug mode: Release
* Specific flags used (if any): clang++ polyhedron_3_to_nef_3_benchmark.cpp -DCGAL_DO_NOT_USE_BOOST_MP -lgmp -lmpfr -lbenchmark
* CGAL version: 5.0.2
* Boost version: 1.71.0
* Other libraries versions if used (Eigen, TBB, etc.): Qt 5.12.8

### PR DESCRIPTION
**Pull request by GilesBathgate**
## Summary of Changes

Implemented the performance improvements as discussed in #5379

* Optimise for Cartesian and Homogeneous kernels
* Use locals instead of multiple functions calls and recalculations
* Use `orthogonal_vector` for triangles

## Release Management

* Affected package(s): Nef_3
* Issue(s) solved (if any): fix #5379
* Feature/Small Feature (if any): performance enhancement.
* License and copyright ownership: Returned to CGAL authors.

### CODE REVIEW

File: Nef_3/include/CGAL/normal_vector_newell_3.h [hunk 1]
```diff
@@ -23,39 +23,83 @@
...
+        * ( py + qy )
+    );
+}
+
+template <class IC>
+bool is_triangle_3( const IC& first )
+{
+    IC last(first);
+    ++last; ++last; ++last;
+    return first==last;
```
Comment (sloriot) [inline Nef_3/include/CGAL/normal_vector_newell_3.h:100]:
```suggestion
    return std::next(first,3) == first;
```

**Review by sloriot — APPROVED** [review]

**Comment by maxGimeno** [comment]
Runtime errors :
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-59/Nef_3/TestReport_lrineau_Debian-Stable.gz

File: Nef_3/include/CGAL/normal_vector_newell_3.h [hunk 2]
```diff
@@ -70,6 +112,14 @@ void normal_vector_newell_3( IC first, IC last, Vector& n )
     // three.
 {
     CGAL_assertion( !CGAL::is_empty_range( first, last));
+
+    if(internal_nef::is_triangle_3(first)) {
+      n = orthogonal_vector(*first,*(std::next(first)),*(std__next(first,2)));
```
Comment (lrineau) [inline Nef_3/include/CGAL/normal_vector_newell_3.h:117]: You should try to compile `std__next` probably does not compile.
```suggestion
      n = orthogonal_vector(*first,*(std::next(first)),*(std::next(first,2)));
```

**Comment by GilesBathgate** [comment]
Thank you @afabri , @lrineau .

**Comment by GilesBathgate** [comment]
Thank you @afabri , @lrineau

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-65.shtml

**Pull request merged**